### PR TITLE
Refactor shorter defaults

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

**Review by jscheffl — APPROVED** [review]

File: airflow/api/common/trigger_dag.py [hunk 1]
```diff
@@ -54,7 +54,8 @@ def _trigger_dag(
     if dag is None or dag_id not in dag_bag.dags:
         raise DagNotFound(f"Dag id {dag_id} not found")
 
-    execution_date = execution_date if execution_date else timezone.utcnow()
+    if execution_date is None:
+        execution_date = timezone.utcnow()
```
Comment (uranusjr) [inline airflow/api/common/trigger_dag.py]: Would `execution_date = execution_date or timezone.utcnow()` work?

Comment (eumiro) [reply airflow/api/common/trigger_dag.py]: Yes. However, while other cases are `c = a or b` and we need to attribute something to `c`, this is `a = a or b`, so if `a` is trueish, it is not modified.

Comment (uranusjr) [reply airflow/api/common/trigger_dag.py]: Yeah and I don’t think that’s a problem for this particular case.

**Comment by vincbeck** [comment]
Conflicts

**Review by uranusjr — APPROVED** [review]

**Review by jscheffl — APPROVED** [review]
CI Failures are from main, otherwise changes are clean. LGTM!

**Comment by potiuk** [comment]
Rebased to see if it works with latest main.

**Review by eladkal — APPROVED** [review]

**Pull request merged**